src/main/java/com/redhat/maven/index/checker/MavenIndexChecker.java

```diff
@@ -267,8 +272,26 @@ public void transferCompleted(TransferEvent transferEvent) {
 
         logger.info("Reading index");
         JSONArray results = new JSONArray();
-        Set<OutputInfo> info = new HashSet<OutputInfo>();
+        File dbFile = File.createTempFile("mic", "db");
+        dbFile.deleteOnExit();
+        /* We use a sqlite on-disk db to be able to leave out duplicates
+           while not keeping all the package objects in-memory.
+           The duplicates exist since the lucene index contains
+           all artifacts for every package version, e.g. not just the
+           binary jar, but also javadocs, sources, ... (if present). */
```

Comment:
The problem here is, that the given gId/aId/version combination can have arbitrary artifacts. It can have sources, but not binary jar, it can only have javadocs, it can have javadocs and binary but no source... I found out no way of expressing this, since I think we want to list every package that has any one of these artifacts uploaded.